autolab/Autolab
# Overwrite labs on reupload

## Summary

### Summary generated by Reviewpad on 29 Jan 24 17:53 UTC
This pull request modifies the `assessments_controller.rb`, `assessment.rb`, `course_logger.rb`, `install_assessment.html.erb`, and `assessments_controller_spec.rb` files.

In `assessments_controller.rb`, the changes include:
- Replacing a check for an existing assessment with a variable assignment.
- Adding a condition to skip certain files based on the existing assessment.
- Adding params for assessment overwrite.

In `assessment.rb`, the changes include:
- Adding methods `asmt_yaml_path`, `log_path`, `unique_source_config_file_path`.
- Modifying methods `dump_yaml` and `load_yaml` to use the new methods.
- Modifying method `path` to handle the new methods.

In `course_logger.rb`, the changes include:
- Modifying method `setAssessment` to use the assessment's log path.

In `install_assessment.html.erb`, the changes include:
- Adding a paragraph explaining the behavior of importing a tarball with the same name as an existing assessment.

In `assessments_controller_spec.rb`, the changes include:
- Modifying the description of a test case.

Overall, these changes seem to improve the management of existing assessments and handle the import of assessment tarballs with the same name.


## Description

- Edit `importAsmtFromTar` to allow duplicate assessment names and upload files for an existing assessment
  - Change `importAssessment` logic to not load any config files if the assessment already exists

## Motivation and Context


Resolves #2017 

## How Has This Been Tested?



Download and unzip these files:
[homework3.tar](https://github.com/autolab/Autolab/files/13614047/homework3.zip)
[homework3_new.tar](https://github.com/autolab/Autolab/files/13614045/homework3_new.zip)

#### Creating your own tar files: ####
1. Export an existing assessment, to get the corresponding tar file, e.g. `homework3.tar`
2. Copy the tar and add another file to the directory, name it smth else like `homework3_new.tar`

#### How to test ####
1. Go to 'Install Assessment' and upload `homework3_new.tar` (don't delete the original asmt)
2. Check that the new file you added to the directory was added and nothing else changed (such as handins/student data)
3. Go to 'Edit assessment' and change the writeup to the new file you added , e.g. `meme.PNG`
4. Save the changes and click 'View writeup', it should download the new file that was uploaded 👍 
5. Go to 'Install Assessment' and upload 'homework3.tar' 
6. Check that the writeup still downloads (ensures that uploading does not delete existing files)


## Types of changes


## Checklist:
